porch/repository/pkg/git/git.go

```diff
@@ -754,3 +779,50 @@ func (r *gitRepository) pushAndCleanup(ctx context.Context, ph *pushRefSpecBuild
 	}
 	return nil
 }
+
+func (r *gitRepository) loadTasks(ctx context.Context, startCommit *object.Commit) ([]v1alpha1.Task, error) {
+	var tasks []v1alpha1.Task
+
+	commitQueue := []*object.Commit{startCommit}
+
+	// Circular refs seem possible
+	done := make(map[plumbing.Hash]bool)
+
+	for {
+		if len(commitQueue) == 0 {
+			break
+		}
+		current := commitQueue[len(commitQueue)-1]
+		commitQueue = commitQueue[:len(commitQueue)-1]
+
+		if done[current.Hash] {
+			continue
+		}
+		done[current.Hash] = true
+
+		klog.Infof("visiting commit %v", current.Hash)
+		for _, line := range strings.Split(current.Message, "\n") {
+			line = strings.TrimSpace(line)
+			if strings.HasPrefix(line, "kpt:") {
+				task := v1alpha1.Task{}
+				b := []byte(strings.TrimPrefix(line, "kpt:"))
+				if err := json.Unmarshal(b, &task); err != nil {
+					klog.Warningf("failed to unmarshal task command %q: %w", line, err)
+					continue
+				}
+				tasks = append(tasks, task)
```

Comment:
if we walk all commits, we will need to filter out commits that belong to different package, to associate tasks with the correct package

... unless we redesign the repository shape and each package gets its own branch. I was hoping to not go there because such repository structure is not what a person using Git to manage packages would create... but maybe they would and we should switch.